lib/src/ovis_json/ovis_json.c

```diff
@@ -541,7 +541,9 @@ int json_attr_rem(json_entity_t d, char *name)
 static void json_list_free(json_list_t a)
 {
 	json_entity_t i;
-	assert(a->base.type == JSON_LIST_VALUE);
+	assert(a && a->base.type == JSON_LIST_VALUE);
```

Comment:
Please

Reply:
do i understand correctly that you want the null check split into a statement separate and before the type assertion, rather than combined? makes sense, just want to double check.